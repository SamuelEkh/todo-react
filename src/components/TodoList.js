import Todo from './Todo';

export default function TodoList(props) {
  const { cards } = props;

  return (
    <section className="card-container">
      {cards.map(card => (
        <Todo
          key={card.id}
          card={card}
          deleteTodo={props.deleteTodo}
          setCompleted={props.setCompleted}
          orderTodos={props.orderTodos} />
      ))}
    </section>
  );
}
