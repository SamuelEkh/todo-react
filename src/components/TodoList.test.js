import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import TodoList from './TodoList';

describe('Starts Empty', () => {
  const cards = [];

  test('Starts empty', () => {
    const { container } = render(<TodoList cards={cards} />);
    expect(container.firstChild.textContent).toEqual('');
  });
});

describe('Several Todos', () => {
  const cards = [
    {
      id: 1,
      title: 'Test Card',
      desc: 'Is this visible?',
      completed: false,
    },
    {
      id: 2,
      title: 'Test Card 2',
      desc: 'Is test card 2 visible?',
      completed: false,
    },
    {
      id: 3,
      title: 'Test Card 3',
      desc: 'Is test card 3 completed?',
      completed: true,
    },
  ];

  const unCompleteTodos = cards.filter(todo => !todo.completed);
  const completeTodos = cards.filter(todo => todo.completed);

  test('unComplete Todos', () => {
    render(<TodoList cards={unCompleteTodos} />);
    const card2Title = screen.getByText('Test Card 2');
    const cardDesc = screen.getByText('Is this visible?');

    expect(() => screen.getByText('Test Card 3')).toThrow();
    expect(card2Title).toBeTruthy();
    expect(cardDesc).toBeTruthy();
  });

  test('Complete Todos', () => {
    render(<TodoList cards={completeTodos} />);
    const finishedCard = screen.getByText('Test Card 3');

    expect(() => screen.getByText('Test Card')).toThrow();
    expect(finishedCard).toBeTruthy();
  });

  it('renders uncomplete correctly', () => {
    const tree = renderer
      .create(<TodoList cards={unCompleteTodos} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
