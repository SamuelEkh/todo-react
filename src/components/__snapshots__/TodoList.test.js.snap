// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Several Todos renders uncomplete correctly 1`] = `
<section
  className="card-container"
>
  <div
    className="card"
  >
    <div
      className="card__content"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      style={
        Object {
          "backgroundColor": "#FFB48F",
          "borderRadius": "5px",
          "boxShadow": "0 0 2px black",
          "display": "flex",
          "flexDirection": "column",
          "height": "7rem",
          "margin": "10px 0",
          "padding": "5px",
          "textAlign": "center",
        }
      }
      tabIndex={0}
    >
      <p
        className="card__title"
      >
        Test Card
      </p>
      <p
        className="card__desc"
      >
        Is this visible?
      </p>
      <p
        className="card__completed"
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0.8)",
            "border": "1px solid black",
            "borderRadius": "5px",
            "color": "green",
            "display": "none",
            "fontSize": "1.5rem",
            "fontWeight": "bold",
            "marginLeft": "5rem",
            "marginTop": "2rem",
            "position": "absolute",
            "transform": "rotate(-25deg)",
          }
        }
      >
        Completed
      </p>
    </div>
    <button
      className="card__button"
      onClick={[Function]}
      style={
        Object {
          "border": "1px solid black",
          "borderRadius": "4px",
          "color": "black",
          "display": "none",
          "margin": "-2.3rem 0 0 0.5rem",
          "position": "absolute",
          "width": "4rem",
        }
      }
      type="button"
    >
      Remove
    </button>
  </div>
  <div
    className="card"
  >
    <div
      className="card__content"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      style={
        Object {
          "backgroundColor": "#FFB48F",
          "borderRadius": "5px",
          "boxShadow": "0 0 2px black",
          "display": "flex",
          "flexDirection": "column",
          "height": "7rem",
          "margin": "10px 0",
          "padding": "5px",
          "textAlign": "center",
        }
      }
      tabIndex={0}
    >
      <p
        className="card__title"
      >
        Test Card 2
      </p>
      <p
        className="card__desc"
      >
        Is test card 2 visible?
      </p>
      <p
        className="card__completed"
        style={
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0.8)",
            "border": "1px solid black",
            "borderRadius": "5px",
            "color": "green",
            "display": "none",
            "fontSize": "1.5rem",
            "fontWeight": "bold",
            "marginLeft": "5rem",
            "marginTop": "2rem",
            "position": "absolute",
            "transform": "rotate(-25deg)",
          }
        }
      >
        Completed
      </p>
    </div>
    <button
      className="card__button"
      onClick={[Function]}
      style={
        Object {
          "border": "1px solid black",
          "borderRadius": "4px",
          "color": "black",
          "display": "none",
          "margin": "-2.3rem 0 0 0.5rem",
          "position": "absolute",
          "width": "4rem",
        }
      }
      type="button"
    >
      Remove
    </button>
  </div>
</section>
`;
